urlchecker/core/worker.py

```diff
@@ -89,7 +87,9 @@ def run(self, funcs, tasks):
             sys.exit(1)
 
         except:
-            logger.exit("Error running task")
+            logger.error("Error running task")
+            pool.terminate()
+            sys.exit(1)
```

Comment:
Ah thanks for catching this! I found it too and we just never merged it: https://github.com/urlstechie/urlchecker-python/pull/64/files#diff-6058668ed83bf6486c58ee5218cce36ef1c25a01e9324033e1a6a3bd8377c703R92

So I think we should do a logger.error here, and allow the worker to return None - the reason is because the caller will see the results are None and then exit with a clear error message. So we don't want to exit too soon here.